### ISSUE TITLE
cabal install silently ignores --enable-profiling flags.

### ISSUE THREAD

**Issue by AndreasPK** [haskell/cabal#8529]
## TLDR

Installing a profiled version of an executable seems to be not supported. 

Instead cabal seems to **silently** swallow the flag.

## **To Reproduce**

This does install a non-profiled version of the executable.
```
andi@horzube:~$ cabal install hasktags --enable-executable-profiling --overwrite-policy=always --enable-profiling
Warning: this is a debug build of cabal-install with assertions enabled.
Resolving dependencies...
Up to date
Warning: installdir is not defined. Set it in your cabal config file or use
--installdir=<path>. Using default installdir: "/home/andi/.cabal/bin"
Symlinking 'hasktags' to '/home/andi/.cabal/bin/hasktags'
andi@horzube:~$ hasktags +RTS -p
hasktags: the flag -p requires the program to be built with -prof
hasktags:

```

I can build a profiled version of a package by running cabal build and copying over, but that seems unneccesary.

**Expected behavior**

I would have expected cabal to install a profiled version of the program if I ask it to. In absence of that I would at least expect it to complain about the flag being ignored/incompatible with install.

But I don't think there is a good reason to have profiling not work with the install command.

**System information**
 - Operating system: Debian
 - `cabal`, `ghc` versions: HEAD/9.4.2

**Additional context**

**Comment by Mikolaj** [comment]
Indeed, `cabal install` ignores some flags (e.g.,  #7236). I didn't know about this one. Thank you for the report.

**Comment by alt-romes** [comment]
This should be fixed by #9697.

**Comment by alt-romes** [comment]
Fixed by #9697
